libs/ledger-live-common/src/families/icon/js-transaction.ts

```diff
@@ -0,0 +1,49 @@
+import { BigNumber } from "bignumber.js";
+import type { IconAccount, Transaction } from "./types";
+
+import getEstimatedFees from "./js-getFeesForTransaction";
+
+const sameFees = (a, b) => (!a || !b ? a === b : a.eq(b));
+
+/**
+ * Create an empty transaction
+ *
+ * @returns {Transaction}
+ */
+export const createTransaction = (): Transaction => ({
+  family: "icon",
+  mode: "send",
+  amount: new BigNumber(0),
+  recipient: "",
+  useAllAmount: false,
+  fees: null,
+});
+
+/**
+ * Apply patch to transaction
+ *
+ * @param {*} t
+ * @param {*} patch
+ */
+export const updateTransaction = (t: Transaction, patch: Transaction) => ({
+  ...t,
+  ...patch,
+});
+
+/**
+ * Prepare transaction before checking status
+ *
+ * @param {IconAccount} a
+ * @param {Transaction} t
+ */
+export const prepareTransaction = async (a: IconAccount, t: Transaction) => {
+  let fees = t.fees;
+
+  fees = await getEstimatedFees({ a, t });
```

Comment:
It seems the "send max" use case isn't handled here?

Note that when the user chooses to do a send max (using the toggle in the UI), `transaction.amount` is _not_ set, `transaction.useAllAmount` is set instead to `true`, and the implementation is responsible for computing the amount.

You can probably call `estimateMaxSpendable` here and use this amount.